### PR TITLE
Add tabs to game list view

### PR DESCRIPTION
**Pull request by eloaverona**
This PR includes some refactoring to allow for better management of the user's public and private keys. It also implements the three different tabs "PLAY", "SPECTATE" and "HISTORY" in the game list view. 

Play Tab
  

Spectate tab

### CODE REVIEW

File: examples/xo_android_client/app/src/main/java/io/bitwise/sawtooth_xo/GameBoardActivity.kt [hunk 1]
```diff
@@ -41,8 +42,4 @@ class GameBoardActivity : AppCompatActivity() {
             super.onOptionsItemSelected(item)
         }
     }
-}
-
-
-
-
+}
```
Comment (dplumb94) [inline examples/xo_android_client/app/src/main/java/io/bitwise/sawtooth_xo/GameBoardActivity.kt]: needs a newline at the end of the file

File: examples/xo_android_client/app/src/main/java/io/bitwise/sawtooth_xo/SharedPreferences.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+private fun generatePrivateKey() : String {
+    val context = Secp256k1Context()
+    return context.newRandomPrivateKey().hex()
+}
+
+private fun generatePublicKey(privateKey: PrivateKey) : String {
+    val context = Secp256k1Context()
+    return context.getPublicKey(privateKey).hex()
+}
```
Comment (dplumb94) [inline examples/xo_android_client/app/src/main/java/io/bitwise/sawtooth_xo/SharedPreferences.kt]: newline

**Review by dplumb94 — CHANGES_REQUESTED** [review]
Two changes:

1. Every file should end in exactly one newline
2. Update filtering based on slack convo

**Review by dplumb94 — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]
tested!

**Pull request merged**